demos/sev-demo/README.md

```diff
@@ -0,0 +1,206 @@
+```
+
+Make sure that the module matches your kernel version. Speaking of which, `efi_secret` also requires some minor kernel patches. These patches and the module are not currently upstream. You can find the v6 patch [here](https://lore.kernel.org/lkml/20211129114251.3741721-1-dovmurik@linux.ibm.com/T/). Build a kernel with this patch and specify it in your Kata configuration file. This will also generate the module that should be specified above.
+
+Once you generate an initrd from the updated rootfs, update your Kata configuration file to point to the new initrd. You will also need to update the kernel field in the configuration file to point to your new kernel. For CCv0 you should have already added `agent.config_file=<agent-config-path>` to the `kernel_params` field. 
+
+## Kata Runtime
+
+Since SEV and SEV-ES use pre-attestation driven from outside of the guest, some changes are required for the Kata shim. These changes are not currently upstream. You will need to build your Kata shim with [this PR](https://github.com/kata-containers/kata-containers/pull/3025). This will work best with QEMU 6.2.0rc0 given that newer version of QEMU will expect to find the `kernel-hashes=on` flag, which is not currently set by the shim.
+
+You will also need to set the following parameters in the Kata configuration file.
+
+First, you'll need to enable SEV via the `confidential_guest` parameter.
+```
+# Enable confidential guest support.
+# Toggling that setting may trigger different hardware features, ranging
+# from memory encryption to both memory and CPU-state encryption and integrity.
+# The Kata Containers runtime dynamically detects the available feature set and
+# aims at enabling the largest possible one.
+# Default false
+confidential_guest = true
+```
+
+Next you'll need to enable pre-attestation.
+```
+# Enable guest attestation
+# (default: false)
+guest_attestation=true
+#
+# Guest owner proxy that handles remote attestation
+guest_attestation_proxy="localhost:50051"
+#
+# Preattestation keyset ID for injected secrets
+guest_attestation_keyset="KEYSET-1"
+```
+Here, we are specifying a local KBS. You can also run the KBS remotely. For this demo we specify the `keyset` in the configuration file. In the future this will be overridden via a Kubernetes annotation. The keyset parameter will be described more below. For now, you can just leave it set to `KEYSET-1`. 
+
+Finally, switch from `virtio-fs` to `virtio-9p`. The former is not currently compatible with SEV. 
+```
+# Shared file system type:
+#   - virtio-fs (default)
+#   - virtio-9p
+shared_fs = "virtio-9p"
+```
+
+## KBS 
+
+You can use the `offline_sev_kbc` manually by injecting a specially formatted JSON file, but you will probably want to use our [key broker server](https://github.com/confidential-containers-demo/scripts/tree/main/guest-owner-proxy) instead. 
+Note that the SEV KBS is often referred to as a Guest Owner Proxy (GOP). This was our original terminology. 
```

Comment:
Good point. We probably should change the name [`gop-*` -> `offline_sev_kbc-*` ] in our demo scripts and kata shim PRs

Reply:
Yeah should probably just ditch the name GOP entirely.